airbyte-bootloader/src/main/java/io/airbyte/bootloader/BootloaderApp.java

```diff
@@ -57,37 +57,36 @@ public BootloaderApp() {
   }
 
   public void load() throws Exception {
-    final Database configDatabase = new ConfigsDatabaseInstance(
-        configs.getConfigDatabaseUser(),
-        configs.getConfigDatabasePassword(),
-        configs.getConfigDatabaseUrl())
-            .getAndInitialize();
-    final DatabaseConfigPersistence configPersistence = new DatabaseConfigPersistence(configDatabase);
-    final ConfigRepository configRepository =
-        new ConfigRepository(configPersistence.withValidation(), null, Optional.empty(), Optional.empty());
-    createWorkspaceIfNoneExists(configRepository);
-    LOGGER.info("Set up config database and default workspace..");
-
-    final Database jobDatabase = new JobsDatabaseInstance(
-        configs.getDatabaseUser(),
-        configs.getDatabasePassword(),
-        configs.getDatabaseUrl())
-            .getAndInitialize();
-    final JobPersistence jobPersistence = new DefaultJobPersistence(jobDatabase);
-    createDeploymentIfNoneExists(jobPersistence);
-    LOGGER.info("Set up job database and default deployment..");
-
-    final AirbyteVersion currAirbyteVersion = configs.getAirbyteVersion();
-    assertNonBreakingMigration(jobPersistence, currAirbyteVersion);
-
-    runFlywayMigration(configs, configDatabase, jobDatabase);
-    LOGGER.info("Ran Flyway migrations...");
-
-    jobPersistence.setVersion(currAirbyteVersion.serialize());
-    LOGGER.info("Set version to {}", currAirbyteVersion);
-
-    configPersistence.loadData(YamlSeedConfigPersistence.getDefault());
-    LOGGER.info("Loaded seed data...");
+    LOGGER.info("Setting up config database and default workspace..");
+
+    try (
+        final Database configDatabase =
+            new ConfigsDatabaseInstance(configs.getConfigDatabaseUser(), configs.getConfigDatabasePassword(), configs.getConfigDatabaseUrl())
+                .getAndInitialize();
+        final Database jobDatabase =
+            new JobsDatabaseInstance(configs.getDatabaseUser(), configs.getDatabasePassword(), configs.getDatabaseUrl()).getAndInitialize()) {
+
+      final DatabaseConfigPersistence configPersistence = new DatabaseConfigPersistence(configDatabase);
+      final ConfigRepository configRepository =
+          new ConfigRepository(configPersistence.withValidation(), null, Optional.empty(), Optional.empty());
+      createWorkspaceIfNoneExists(configRepository);
+
+      final JobPersistence jobPersistence = new DefaultJobPersistence(jobDatabase);
+      createDeploymentIfNoneExists(jobPersistence);
+      LOGGER.info("Set up job database and default deployment..");
+
+      final AirbyteVersion currAirbyteVersion = configs.getAirbyteVersion();
+      assertNonBreakingMigration(jobPersistence, currAirbyteVersion);
+
+      runFlywayMigration(configs, configDatabase, jobDatabase);
```

Comment:
As discussed running flyway migration should be the first step before we make any interaction with the database. We are creating a workspace, and a deployment above before executing the migration but as in my config migration, the DatabaseConfigPersistence will try to interact with `workspace` table which has not been created yet.

Reply:
good point!

Reply:
@subodh1810 how is this?